airbyte-integrations/connectors/destination-google-sheets/destination_google_sheets/writer.py

```diff
@@ -34,12 +34,13 @@ def queue_write_operation(self, stream_name: str):
         """
         Mimics `batch_write` operation using records_buffer.
 
-        1) gets data from the records_buffer
+        1) gets data from the records_buffer, with respect to the size of the records_buffer (records count or size in Kb)
         2) writes it to the target worksheet
         3) cleans-up the records_buffer belonging to input stream
         """
-
-        if len(self.records_buffer[stream_name]) == self.flush_interval:
+        # get the size of records_buffer for target stream in Kb
+        records_buffer_size_in_kb = self.records_buffer[stream_name].__sizeof__() / 1024
```

Comment:
not blocking this PR over this since it's on neither of y'alls backlogs right now and is a "drive-by" change
```suggestion
        # TODO unit test flush triggers
        records_buffer_size_in_kb = self.records_buffer[stream_name].__sizeof__() / 1024
```